src/components/Portfolio/TotalBalance.tsx

```diff
@@ -0,0 +1,206 @@
+  const [differenceValue, setDifferenceValue] = useState(0);
+  const [differencePercent, setDifferencePercent] = useState(0);
+  const { t } = useTranslation();
+  const { accounts } = useAccount();
+  const portfolio = usePortfolioStore((state) => state);
+  const { totalValue } = portfolio.getFormattedCacheTokens(accounts);
+
+  useEffect(() => {
+    if (!isComplete) {
+      return;
+    }
+
+    const addresses = accounts
+      .filter((a) => !!a?.address)
+      .map(({ address }) => address) as string[];
+
+    if (addresses.length === 0) {
+      return;
+    }
+
+    if (!arraysEqual(portfolio.lastAddresses ?? [], addresses)) {
+      portfolio.setLast(totalValue, addresses);
+      return;
+    }
+
+    if (portfolio.lastDate && !has24HoursPassed(portfolio.lastDate)) {
+      return;
+    }
+
+    portfolio.setLast(totalValue, addresses);
+
+    if (!portfolio.lastTotalValue) {
+      return;
+    }
+
+    const differenceValue = totalValue - portfolio.lastTotalValue;
+    const differencePercent =
+      portfolio.lastTotalValue !== 0
+        ? ((totalValue - portfolio.lastTotalValue) /
+            Math.abs(portfolio.lastTotalValue)) *
+          100
+        : 0;
+
+    setDifferenceValue(differenceValue);
+    setDifferencePercent(differencePercent);
+  }, [totalValue, isComplete]);
+
+  if (!isComplete && totalValue === 0) {
+    return <TotalBalanceSkeleton />;
+  }
+
+  return (
+    <WalletCardContainer>
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          alignContent: 'center',
+          alignItems: 'center',
+        }}
+      >
+        <Box
+          sx={{
+            display: 'flex',
+            justifyContent: 'center',
+            alignContent: 'center',
+            alignItems: 'center',
+          }}
+        >
+          <Typography
+            fontWeight={500}
+            fontSize={14}
+            color={(theme) => theme.palette.text.primary}
+          >
+            {t('navbar.walletMenu.totalBalance')}
+          </Typography>
+          <Tooltip
+            title={t('navbar.walletMenu.totalBalanceTooltip')}
+            placement="top"
+            enterTouchDelay={0}
+            componentsProps={{
+              popper: { sx: { zIndex: 2000 } },
+            }}
+            arrow
+            sx={{
+              zIndex: 25000,
+            }}
+          >
+            <InfoIcon
+              sx={{
+                cursor: 'help',
+                marginLeft: '8px',
+                width: 16,
+                height: 16,
+                opacity: '0.5',
+              }}
+            />
+          </Tooltip>
+        </Box>
+        {!isComplete ? (
+          <TotalBalanceIconButton disabled={true}>
+            <CircularProgressPending size={24} />
+          </TotalBalanceIconButton>
+        ) : (
+          <TotalBalanceIconButton
+            tooltipText="Click here to restart the indexing of your tokens now."
```

Comment:
We have other parts inside translation. Maybe we could also add this?

Reply:
I will.